pandas/tests/plotting/frame/test_hist_box_by.py

```diff
@@ -0,0 +1,289 @@
+                    "(a, c)",
+                    "(b, a)",
+                    "(b, b)",
+                    "(b, c)",
+                    "(c, a)",
+                    "(c, b)",
+                    "(c, c)",
+                ],
+                [["A", "B"]] * 9,
+            ),
+            (
+                ["C", "D"],
+                None,
+                [
+                    "(a, a)",
+                    "(a, b)",
+                    "(a, c)",
+                    "(b, a)",
+                    "(b, b)",
+                    "(b, c)",
+                    "(c, a)",
+                    "(c, b)",
+                    "(c, c)",
+                ],
+                [["A", "B"]] * 9,
+            ),
+        ],
+    )
+    def test_hist_plot_by_argument(self, by, column, titles, legends):
+        # GH 15079
+        axes = _check_plot_works(self.hist_df.plot.hist, column=column, by=by)
+        result_titles = [ax.get_title() for ax in axes]
+        result_legends = [
+            [legend.get_text() for legend in ax.get_legend().texts] for ax in axes
+        ]
+
+        assert result_legends == legends
+        assert result_titles == titles
+
+    @pytest.mark.slow
+    @pytest.mark.parametrize(
+        "by, column, layout, axes_num",
+        [
+            (["C"], "A", (2, 2), 3),
+            ("C", "A", (2, 2), 3),
+            (["C"], ["A"], (1, 3), 3),
+            ("C", None, (3, 1), 3),
+            ("C", ["A", "B"], (3, 1), 3),
+            (["C", "D"], "A", (9, 1), 9),
+            (["C", "D"], "A", (3, 3), 9),
+            (["C", "D"], ["A"], (5, 2), 9),
+            (["C", "D"], ["A", "B"], (9, 1), 9),
+            (["C", "D"], None, (9, 1), 9),
+            (["C", "D"], ["A", "B"], (5, 2), 9),
+        ],
+    )
+    def test_hist_plot_layout_with_by(self, by, column, layout, axes_num):
+        # GH 15079
+        # _check_plot_works adds an ax so catch warning. see GH #13188
+        with tm.assert_produces_warning(UserWarning):
+            axes = _check_plot_works(
+                self.hist_df.plot.hist, column=column, by=by, layout=layout
+            )
+        self._check_axes_shape(axes, axes_num=axes_num, layout=layout)
+
+    def test_hist_plot_invalid_layout_with_by_raises(self):
+        # GH 15079, test if error is raised when invalid layout is given
+
+        # layout too small for all 3 plots
+        msg = "larger than required size"
+        with pytest.raises(ValueError, match=msg):
+            self.hist_df.plot.hist(column=["A", "B"], by="C", layout=(1, 1))
+
+        # invalid format for layout
+        msg = re.escape("Layout must be a tuple of (rows, columns)")
+        with pytest.raises(ValueError, match=msg):
+            self.hist_df.plot.hist(column=["A", "B"], by="C", layout=(1,))
+
+        msg = "At least one dimension of layout must be positive"
+        with pytest.raises(ValueError, match=msg):
+            self.hist_df.plot.hist(column=["A", "B"], by="C", layout=(-1, -1))
```

Comment:
No big deal, but maybe we could parametrize this?

Reply:
Sure! Done!